### ISSUE TITLE
add new evolutions

### ISSUE THREAD

**Issue by RSikkel** [mtaheij/PoGo-UWP#41]
The second generations have new evolutions, some of which need special items (such as onyx --> steelix). Would love to see them in the app! Slightly related note, after having spun hundreds of pokestops I still never received a special item. Are they implemented yet?

**Comment by ntaheij** [comment]
Yea. I got one last time, but the change that you get them, is really low

**Comment by andremvm** [comment]
I got my first special item today, but I don't know exactly how/why. It was in a pokestop where I pass everyday and gives me usually just 2-3 items; today however I got a mix of 16 pokeballs, a bunch of potions, and also the King's Rock!

**Comment by RSikkel** [comment]
It's probably for your seven day streak (at least one pokestop every day for seven days). Allegedly, the special items are a more common occurrence for these spins. Got my first special item today as well! Looking forward to being able to use it.

**Comment by andremvm** [comment]
I don't know if that is quite true. Except for the days where the app was down I have always caught at least 1 pokemon and spun 1 pokestop per day to make sure I would always get the bonus points! :P But let's see if it keeps happening... :)

**Comment by mtaheij** [comment]
It's true: http://pokemongolive.com/en/post/ver-update-032117

**Comment by andremvm** [comment]
Oh, but then it is something that it was announced just yesterday! Great news then :)

**Comment by mtaheij** [comment]
Could be that they've tested it before, because I have seen the special evolution item last week with my own eyes :-)

**Comment by BadiBasso** [comment]
Pokestops do drop evo items but they are really few and far between. Had one a couple a weeks ago and another one yesterday :)
Neither were from 7 day streak.

**Comment by DarkLegend481** [comment]
Yeah... After 169 PokéStops (I use bot... My mom don't let me out of home to catch Pokémon... Kkkkkk), I got a Dragon Scale! Now, I want a Kingdra!

**Comment by andremvm** [comment]
Well, I got the same item last Wednesday (once again, the 7-day streak), in the first and only pokéstop of the day... :P

**Comment by RSikkel** [comment]
Unrelated comment: Lost my phone on wintersport holliday, so now I have an Android phone. If the developers need any information about how the official app behaves or what it looks like, let me know :)

Not sure whether it happened because of my pogo uwp usage after the last update or before, but I do get the 'we detected unusual behavior' screen in the official app. Won't go away so far either.

**Comment by mtaheij** [comment]
In the currently used protos (internal API fields sort of say), there is now a 'banned' flag and a 'warn' flag. I think the last one means that your account is flagged and thus in the original app the 'we detected unusual behavior' screen appears. Don't know if this flag will remain active once it has been once.
As long as the 'banned' flag is not set to true, we're ok, I guess :-)

**Comment by BadiBasso** [comment]
I'm currently working on the new evolutions. Some screenshots of how they are implemented would be nice.

**Comment by andremvm** [comment]
My sister sent me this a few days ago because she wasn't aware of the new evolutions and wanted to know what it was: 

![17106231_10208871129277761_1746956453_o](https://cloud.githubusercontent.com/assets/25388446/24553420/8295e452-162a-11e7-9e2f-bc8c53bd26c2.png)

**Comment by BadiBasso** [comment]
Have three different (six in total) evo items, so that really pushes me forward. ;)

**Comment by BadiBasso** [comment]
Thanks! That's the one occasion where you can choose evo line. Was wondering how that was handled.

**Comment by RSikkel** [comment]
![screenshot_20170331-160313](https://cloud.githubusercontent.com/assets/22431510/24554022/76e8a05c-162c-11e7-99e8-443e4681c444.png)

The battle screen changed somewhat as well

**Comment by mtaheij** [comment]
@RSikkel: Can you place this last (battle/encounter) image in a separate issue please? This issue is for evolutions.

**Comment by mtaheij** [comment]
@BadiBasso In the GameClient class, you can change the EvolvePokemon method like this
![evolvepokemonmethod](https://cloud.githubusercontent.com/assets/769923/24554821/3b9cebae-162f-11e7-8671-690fe7943994.PNG)

**Comment by BadiBasso** [comment]
@mtaheij Thanks, I know the new call.
Main issue here is that evo lines are now defined in two places in protos. And the current implementation in pogo-uwp is quite cumbersome. Trying to get it more smooth and easier to update when/if new gens and evo lines are introduced.

**Comment by BadiBasso** [comment]
Current implementation can not handle the two evos of gloom, for example.

**Comment by BadiBasso** [comment]
Seems I got it working :)
Will try to create a PR this weekend.
![wp_ss_20170428_0001](https://cloud.githubusercontent.com/assets/25418803/25529779/3c2439c0-2c2c-11e7-9640-deea463356f8.png)

![wp_ss_20170428_0003](https://cloud.githubusercontent.com/assets/25418803/25529794/47ba3d02-2c2c-11e7-8e89-082a46bf6ef6.png)
